### PR TITLE
Use react-is instead of typeof on getTag

### LINKED ISSUE #221: Cannot pass React.forwardRef directly
I'm used to write overrides like this:
```js
const overrides = {
  h1: H1,
  h2: H2,
}
```

But since I started wrapping my components with `React.forwardRef` it didn't work anymore. I have to use the object version.

Reproduction: https://codesandbox.io/s/p7w1o7p4yx

### PR DESCRIPTION
**Pull request by diegohaz**
Fixes #221

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -63,6 +63,7 @@
   },
   "dependencies": {
     "prop-types": "^15.5.10",
+    "react-is": "^16.5.2",
```
Comment (probablyup) [inline package.json]: Can you move this to peerDeps?

Comment (diegohaz) [reply package.json]: Sure. But it would be a breaking change, wouldn't it?

**Comment by probablyup** [comment]
Yup. It'll go out with v7

**Comment by probablyup** [comment]
I ended up fixing this a different way that doesn't need a major version bump :) 7258f196fd91e3bfab4df9dd948c8451f2e63990

**Pull request closed without merging**